src/client.ts

```diff
@@ -116,34 +154,99 @@ export class APIClient {
    * @param iteratee receives each resource to produce entities/relationships
    */
   public async iterateGroups(
-    iteratee: ResourceIteratee<AcmeGroup>,
+    iteratee: ResourceIteratee<ADOGroup>,
   ): Promise<void> {
-    // TODO paginate an endpoint, invoke the iteratee with each record in the
-    // page
-    //
-    // The provider API will hopefully support pagination. Functions like this
-    // should maintain pagination state, and for each page, for each record in
-    // the page, invoke the `ResourceIteratee`. This will encourage a pattern
-    // where each resource is processed and dropped from memory.
-
-    const groups: AcmeGroup[] = [
-      {
-        id: 'acme-group-1',
-        name: 'Group One',
-        users: [
-          {
-            id: 'acme-user-1',
-          },
-        ],
-      },
-    ];
+    const groups: ADOGroup[] = [];
+
+    try {
+      const authHandler = azdev.getPersonalAccessTokenHandler(
+        this.config.accessToken,
+      );
+      const connection = new azdev.WebApi(this.config.orgUrl, authHandler);
+      const core: cr.ICoreApi = await connection.getCoreApi();
+      const allTeams = await core.getAllTeams();
+      for (const team of allTeams) {
+        if (team.projectId != undefined && team.id != undefined) {
+          const group: ADOGroup = team;
+          group.users = [];
+          const teamMembers = await core.getTeamMembersWithExtendedProperties(
+            team.projectId,
+            team.id,
+          );
+          for (const teamMember of teamMembers) {
+            if (teamMember.identity?.id != undefined) {
```

Comment:
```suggestion
            if (teamMember.identity?.id) {
```
This is sufficient, unless `id` could be `0`. If the comparison is required, `!==` does an identity check (instead of equality) which is best for `undefined` comparisons.

Reply:
I don't know if there is a chance of `teamMember` having an `id` of 0. It seems unlikely, but just in case, I went with `!==` here. On most of the other cases (like project id and team id) I just simplified the if condition per your indication here.